magenta-lib/src/test/scala/magenta/tasks/ASGTest.scala

```diff
@@ -266,133 +266,138 @@ class ASGTest extends AnyFlatSpec with Matchers with MockitoSugar {
-
-    ASG.isStabilized(
-      group,
-      ELB.Client(classicELBClient, appELBClient)
-    ) shouldBe Right(())
-  }
-
-  it should "wait for classic ELB and ELB target group to report as healthy" in {
-    val appELBClient = mock[ApplicationELBClient]
-    val classicELBClient = mock[ClassicELBClient]
-
-    val group = AutoScalingGroupWithTags(
-      "elb",
-      "Role" -> "example",
-      "Stage" -> "PROD"
-    ).toBuilder
-      .targetGroupARNs("elbTargetARN")
-      .desiredCapacity(1)
-      .build()
-
-    mockTargetGroupHealth(appELBClient, TargetHealthStateEnum.UNHEALTHY)
-
-    mockClassicELBGroupHealth(classicELBClient, "")
-
-    ASG.isStabilized(group, ELB.Client(classicELBClient, appELBClient)) shouldBe
-      Left("Only 0 of 1 Classic ELB instances InService")
-
-    mockClassicELBGroupHealth(classicELBClient, "InService")
-
-    ASG.isStabilized(group, ELB.Client(classicELBClient, appELBClient)) shouldBe
-      Left("Only 0 of 1 V2 ELB instances healthy")
-
-    mockTargetGroupHealth(appELBClient, TargetHealthStateEnum.HEALTHY)
-
-    ASG.isStabilized(
-      group,
-      ELB.Client(classicELBClient, appELBClient)
-    ) shouldBe Right(())
-  }
-
-  it should "just check ASG health for stability if there is no ELB" in {
-    val appELBClient = mock[ApplicationELBClient]
-    val classicELBClient = mock[ClassicELBClient]
-
-    val group =
-      AutoScalingGroupWithTags("Role" -> "example", "Stage" -> "PROD").toBuilder
-        .desiredCapacity(1)
-        .instances(ASGInstance.builder().healthStatus("Foobar").build())
-        .build()
-
-    when(
-      classicELBClient.describeInstanceHealth(
-        DescribeInstanceHealthRequest
-          .builder()
-          .loadBalancerName("elb")
-          .build()
-      )
-    ).thenReturn(
-      DescribeInstanceHealthResponse
-        .builder()
-        .instanceStates(
-          InstanceState
-            .builder()
-            .state("")
-            .build()
-        )
-        .build()
-    )
-
-    ASG.isStabilized(group, ELB.Client(classicELBClient, appELBClient)) shouldBe
-      Left("Only 0 of 1 instances InService")
-
-    val updatedGroup =
-      AutoScalingGroupWithTags("Role" -> "example", "Stage" -> "PROD").toBuilder
-        .desiredCapacity(1)
-        .instances(
-          ASGInstance
-            .builder()
-            .lifecycleState(LifecycleState.IN_SERVICE)
-            .build()
-        )
-        .build()
-
-    ASG.isStabilized(
-      updatedGroup,
-      ELB.Client(classicELBClient, appELBClient)
-    ) shouldBe Right(())
-  }
+  // TODO: these tests regularly fail with `An illegal reflective access
+  // operation has occurred`. This appears to be caused by Mockito when mocking
+  // the `ApplicationELBClient` and/or `ClassicELBClient` classes. We should
+  // investigate this further and re-enable these tests once fixed.œ
+
+//  it should "wait for instances in ELB to stabilise if there is one" in {
```

Comment:
Could we use the `@Ignore` annotation? It'll mean we'll be reminded to fix the test in the log (both in CI and locally).

Reply:
👍 agreed, much better than just commenting out.